src/components/Reseve/Reserve.jsx

```diff
@@ -0,0 +1,460 @@
+import React, { useState, useEffect } from 'react';
+import './Reserve.css';
+import { Link, useParams } from 'react-router-dom';
+import { useDispatch, useSelector } from 'react-redux';
+import Dialog from '../Dialog/Dialog';
+import rectangle from '../../Assets/rectangle.png';
+import {
+  reserveActions,
+  selectReserveError,
+} from '../../features/slices/reserve/reserveSlice';
+import {
+  fetchHotel,
+  selectSingleHotel,
+  selectSingleHotelLoading,
+} from '../../features/slices/reserve/singleReserveSlice';
+import { readRooms } from '../../features/slices/roomTypes/fetchRooms';
+
+function Reserve() {
```

Comment:
- [ ] In the reservation page,  It claims that there are no hotels for me to reserve:

![image](https://github.com/felixDev22/Final-Capstone-Frontend/assets/57421823/c76b91e5-d1f3-4fb9-9f26-ff86c940c8ed)

But on the main page, it's full of hotels:

![image](https://github.com/felixDev22/Final-Capstone-Frontend/assets/57421823/3e09c81d-cad6-4100-8990-f59a810ef87e)

And when I reserve a hotel from the main page, the reservation appears on the same page that claimed that I cannot reserve:

![image](https://github.com/felixDev22/Final-Capstone-Frontend/assets/57421823/8a1347f9-4f13-46a3-b625-4515ac35506d)


Kindly take a look at this, so users can reserve from that page. 👍